File: ProductPurchaseApp/ProductPurchaseApp.Web/Modules/ProductPurchaseRequestDB/ProductPurchaseRequest/ProductPurchaseRequestColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace ProductPurchaseApp.ProductPurchaseRequestDB.Columns;

[ColumnsScript("ProductPurchaseRequestDB.ProductPurchaseRequest")]
[BasedOnRow(typeof(ProductPurchaseRequestRow), CheckNames = true)]
public class ProductPurchaseRequestColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PurchaseRequestId { get; set; }
    public int UserId { get; set; }
    [EditLink]
    public string ProductName { get; set; }
    public string Status { get; set; }
    public string Attachment { get; set; }
    public DateTime CreatedDate { get; set; }
}

File: ProductPurchaseApp/ProductPurchaseApp.Web/Modules/ServerTypes/ProductPurchaseRequestDB/ProductPurchaseRequestColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductPurchaseRequestRow } from "./ProductPurchaseRequestRow";

export interface ProductPurchaseRequestColumns {
    PurchaseRequestId: Column<ProductPurchaseRequestRow>;
    UserId: Column<ProductPurchaseRequestRow>;
    ProductName: Column<ProductPurchaseRequestRow>;
    Status: Column<ProductPurchaseRequestRow>;
    Attachment: Column<ProductPurchaseRequestRow>;
    CreatedDate: Column<ProductPurchaseRequestRow>;
}

export class ProductPurchaseRequestColumns extends ColumnsBase<ProductPurchaseRequestRow> {
    static readonly columnsKey = 'ProductPurchaseRequestDB.ProductPurchaseRequest';
    static readonly Fields = fieldsProxy<ProductPurchaseRequestColumns>();
}